### PR TITLE
Add FormatProvider for FXCop error CA1305. Fix failing tests and a bug

### PR DESCRIPTION
**Pull request by pieandcakes**
Once I fixed the format provider, the SSH PS tests were failing, which showed a bug. Updated tests with new PS format and fixed the bug.

### CODE REVIEW

**Comment by pieandcakes** [comment]
@WardenGnaw @gregg-miskelly Apparently some test are failing too so I need to fix those also.

**Comment by pieandcakes** [comment]
@gregg-miskelly @WardenGnaw Can you take another look. This should pass the build now.

File: src/SSHDebugPS/PSOutputParser.cs [hunk 1]
```diff
@@ -147,7 +147,7 @@ private bool ProcessHeaderLine(/*OPTIONAL*/ string headerLine)
             SkipNonWhitespace(headerLine, ref index);
 
             // make sure the line is now empty, aside from whitespace
-            if (SkipWhitespace(headerLine, ref index))
+            if (SkipWhitespace(headerLine, ref index) && index < strLen)
```
Comment (gregg-miskelly) [inline src/SSHDebugPS/PSOutputParser.cs]: Why did you add the condition? `SkipWhitespace` will return false if there is no more whitespace before the end.

File: src/SSHDebugPS/SSH/SSHConnection.cs [hunk 2]
```diff
@@ -46,7 +46,7 @@ public override List<Process> ListProcesses()
             var usernameCommand = _remoteSystem.Shell.ExecuteCommand("id -u -n");
             if (usernameCommand.ExitCode == 0)
             {
-                username = usernameCommand.Output;
+                username = usernameCommand.Output.TrimEnd('\n'); // trim line endings because 'id' command ends with a newline
```
Comment (gregg-miskelly) [inline src/SSHDebugPS/SSH/SSHConnection.cs]: Should we trim `\r` also?

Comment (pieandcakes) [reply src/SSHDebugPS/SSH/SSHConnection.cs]: I can add it. Will we have '\r' on Linux?
